cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/bitwarden"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	logLevel       string
+	configPath     string
+	bwUser         string
+	dryRun         bool
+	bwPasswordPath string
+	maxConcurrency int
+
+	config     []bitWardenItem
+	bwPassword string
+}
+
+type bitWardenItem struct {
+	ItemName   string         `json:"item_name"`
+	Field      fieldGenerator `json:"field,omitempty"`
```

Comment:
The `omitempty` doesn't do anything on literal structs, only on pointers: https://stackoverflow.com/questions/18088294/how-to-not-marshal-an-empty-struct-into-json-with-go

Reply:
Ok. I am removing the omitempty. Since I have checks for stringempty on the names fields before using them